cluster-autoscaler/processors/status/scale_down_status_processor.go

```diff
@@ -61,6 +61,36 @@ const (
 	ScaleDownInProgress
 )
 
+// NodeDeleteResultType denotes the type of the result of node deletion. It provides deeper
+// insight into why the node failed to be deleted.
+type NodeDeleteResultType int
+
+const (
+	// NodeDeleteOk - the node was deleted successfully.
+	NodeDeleteOk NodeDeleteResultType = iota
+
+	// NodeDeleteErrorFailedToMarkToBeDeleted - node deletion failed because the node couldn't be marked to be deleted.
+	NodeDeleteErrorFailedToMarkToBeDeleted
+	// NodeDeleteErrorFailedToEvictPods - node deletion failed because some of the pods couldn't be evicted from the node.
+	NodeDeleteErrorFailedToEvictPods
+	// NodeDeleteErrorPodsFailedToDisappear - node deletion failed because some of the pods didn't disappear from the node in time after eviction.
+	NodeDeleteErrorPodsFailedToDisappear
+	// NodeDeleteErrorFailedToDelete - failed to delete the node from the cloud provider.
+	NodeDeleteErrorFailedToDelete
+)
+
+// NodeDeleteResult contains information about the result of a node deletion.
+type NodeDeleteResult struct {
+	// Err contains nil if the delete was successful and an error otherwise.
+	Err error
+	// ResultType contains the type of the result of a node deletion.
+	ResultType NodeDeleteResultType
+	// FailingPods contains the pods that prevented the node from being deleted, if there were any.
```

Comment:
Just an idea: you already create podEvictionResult type, why not move it here and replace both 'FailingPods' and 'FailingPodErrors' with:

PodEvictionResults map[string]PodEvictionResult

For pods that still exist, attach a new error to result. Then the processors can deal with this to aggregate etc.

Reply:
Done, something similar at least. Looks much better!